File: Daily/sbgo/Leetcode206-反转链表.go
package sbgo

func reverseList(head *ListNode) *ListNode {
    var prev *ListNode
    cur := head
    if head == nil{
        return head
    }
    for cur != nil{
        tmp := cur.Next
        cur.Next = prev
        prev = cur
        cur = tmp
    }
    return prev
}

File: Daily/PY/Leetcode206-反转链表.py
from typing import List

class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


class Solution:
    def reverseList(self, head: ListNode) -> ListNode:
        prev = None
        cur = head
        if head == None:
            return head
        while cur != None:
            tmp = cur.next
            cur.next = prev
            prev = cur 
            cur = tmp
        return prev 